src/source/Signaling/Signaling.c

```diff
@@ -364,6 +363,29 @@ STATUS signalingGetIceConfigInfo(PSignalingClient pSignalingClient, UINT32 index
     return retStatus;
 }
 
+STATUS signalingFetchSync(PSignalingClient pSignalingClient)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(pSignalingClient != NULL, STATUS_NULL_ARG);
+
+    // Check if we are already not connected
+    if (ATOMIC_LOAD_BOOL(&pSignalingClient->connected)) {
+        CHK_STATUS(terminateOngoingOperations(pSignalingClient));
+    }
+
+    // move to the fromGetToken() so we can move to the necessary step
+    setStateMachineCurrentState(pSignalingClient->pStateMachine, SIGNALING_STATE_GET_TOKEN);
```

Comment:
Could be a stretch, but want to make sure we are covering all basis and not missing any data corruption. If GET_TOKEN succeeds in the `createSignalingClient()` call, the service result would be `SERVICE_CALL_RESULT_OK` which is what we check for in `fromGetTokenSignalingState` while deciding state transitions. Just need to ensure there is not a weird scenario where the `pSignalingClient->result` changes to an error service code between `connect` and `fetch` calls, which might lead to the `GET_TOKEN` call being made again.

Reply:
That result should only change from those exact calls. No opportunity to change it between them exists.